(release_30)bugfix: wrong ID number Data-type

SlySven opened the pull request:
Although the Tree.h uses qint64 for the (internal) ID numbers of actions,
aliases, keys, scripts, timers and triggers the rest of the code base uses
an int data-type instead.  This commit changes everything to use "int"
though perhaps it may have been better to explicitly use the Qt qint32 type
that would require careful examination of a LOT of code to check every
location was using the explicit type. As it is this mismatch is not likely
to cause problems as no-one is going to exceed the 2,147,483,647 limit that
using an int type (signed 32-bits) that the likely platforms will use
although strictly speaking the C99 standard only mandates a signed int
maximum value of 32,767 using 16 bits...!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented:
Looks like our Travis is bugged.

On Fri, 24 Jun 2016 2:17 am Stephen Lyons, <notifications@github.com> wrote:

> Although the Tree.h uses qint64 for the (internal) ID numbers of actions,
> aliases, keys, scripts, timers and triggers the rest of the code base uses
> an int data-type instead. This commit changes everything to use "int"
> though perhaps it may have been better to explicitly use the Qt qint32 type
> that would require careful examination of a LOT of code to check every
> location was using the explicit type. As it is this mismatch is not likely
> to cause problems as no-one is going to exceed the 2,147,483,647 limit that
> using an int type (signed 32-bits) that the likely platforms will use
> although strictly speaking the C99 standard only mandates a signed int
> maximum value of 32,767 using 16 bits...!
>
> Signed-off-by: Stephen Lyons slysven@virginmedia.com
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/Mudlet/Mudlet/pull/312
> Commit Summary
>
>    - Bugfix: wrong ID number Data-type
>
> File Changes
>
>    - *M* src/ActionUnit.cpp
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-0> (2)
>    - *M* src/ActionUnit.h
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-1> (4)
>    - *M* src/AliasUnit.cpp
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-2> (2)
>    - *M* src/AliasUnit.h
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-3> (12)
>    - *M* src/KeyUnit.cpp
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-4> (2)
>    - *M* src/KeyUnit.h
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-5> (4)
>    - *M* src/ScriptUnit.cpp
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-6> (2)
>    - *M* src/ScriptUnit.h
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-7> (4)
>    - *M* src/TAction.h
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-8> (2)
>    - *M* src/TTimer.cpp
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-9> (4)
>    - *M* src/TTimer.h
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-10> (4)
>    - *M* src/TimerUnit.cpp
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-11> (2)
>    - *M* src/TimerUnit.h
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-12> (4)
>    - *M* src/Tree.h
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-13> (6)
>    - *M* src/TriggerUnit.cpp
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-14> (2)
>    - *M* src/TriggerUnit.h
>    <https://github.com/Mudlet/Mudlet/pull/312/files#diff-15> (4)
>
> Patch Links:
>
>    - https://github.com/Mudlet/Mudlet/pull/312.patch
>    - https://github.com/Mudlet/Mudlet/pull/312.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Mudlet/Mudlet/pull/312>, or mute the thread
> <https://github.com/notifications/unsubscribe/AAGxjBYzFoRjKyrmJupWAjmmNZcsCDKXks5qOrGVgaJpZM4I8_Df>
> .
>

SlySven commented:
:bomb: Oh b****r - according to the details on https://launchpad.net/~ubuntu-sdk-team/+archive/ubuntu/ppa the Ubunta 12.04 LTS Qt 5.0 packages have been move elsewhere - and the Travis CI is failing on ALL Linux build as that uses Qt 5.0.2 (so that we can ensure that we have coverage for all versions of Qt 5.x) - in comparison the MacOs builds use the latest bleeding edge Qt 5.x - however there was some Travis internal problems earlier on today (yesturday) for MacOs builds and jobs had been put on hold.

:hammer: I have (re)-kicked the two of four failing MacOs builds but they are taking an age to boot-up...!  I am looking to see how we can grab those relocated Ubuntu Linux 12.04 LTS packages back into the build - OK Done that and it is working now for the Linux builds.

Still have a (second) issue in that on the MacOs platform the CMake builds are failing (for both GCC and Clang compilers) because there is a failure to detect the Hunspell Libraries (the FindHUNSPELL.cmake "finder" is no longer working to set the `HUNSPELL_LIBRARIES` variable) as a result of the:
`FIND_PACKAGE(HUNSPELL REQUIRED)` at _line 348_ in our `./src/CMakeLists.txt` which calls:
`FIND_PACKAGE_HANDLE_STANDARD_ARGS(HUNSPELL  DEFAULT_MSG  HUNSPELL_LIBRARIES HUNSPELL_INCLUDE_DIR)` at _line 74_ of `./cmake/FindHUNSPELL.cmake`. __Ah ha! :bulb: it seems that using packageConfig to detect the relevant directories (as we do already for some of the other needed libraries on the MacOs platform seems to restore the previous successful build)__ _Just running a final check (on the development branch), if that is okay I'll append the squashed down commits that modify the C.I. setup files on to this branch as well..._

SlySven commented:
OK, have re-based (squashed together) the fixes for the Travis C.I. here and duplicated the net changes onto the development branch #311 so they should both now build to the same level of success (7 - combinations compile OK, 1 - Linux/Cmake/Clang fails due to a internal Qt 5.0.2 issue) as we previously experienced... :sunglasses:

SlySven commented on src/CMakeLists.txt:
```diff
@@ -333,6 +333,23 @@ IF((APPLE) AND (PKG_CONFIG_FOUND))
+    IF(PC_HUNSPELL_hunspell_FOUND)
+      MESSAGE(STATUS "Using pkg_config, found \"hunspell\" version: ${PC_HUNSPELL_hunspell_VERSION} with:")
+    ELSEIF(PC_HUNSPELL_libhunspell_FOUND)
+      MESSAGE(STATUS "Using pkg_config, found \"libzzip\" version: ${PC_HUNSPELL_libhunspell_VERSION} with:")
```
Woops - forgot to edit that copied text - should of course read "... found \"libhunspell\" version..." but it is only a reporting text and the script works... :blush:

SlySven commented:
OK, fixed the CMake advisory message mentioned in the previous comment - and removed a _FIXME: warning_ in the top level `CMakeList.txt` file that is not longer relevant as the thing it referred to has been FIXed...! :smile:

@vadi2  May I merge in this and PR #311?

vadi2 commented:
+1

On Sat, Jul 2, 2016 at 2:29 AM Stephen Lyons <notifications@github.com>
wrote:

> OK, fixed the CMake advisory message mentioned in the previous comment -
> and removed a *FIXME: warning* in the top level CMakeList.txt file that
> is not longer relevant as the thing it referred to has been FIXed...! 😄
>
> @vadi2 <https://github.com/vadi2> May I merge in this and PR #311
> <https://github.com/Mudlet/Mudlet/pull/311>?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/Mudlet/Mudlet/pull/312#issuecomment-229990630>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AAGxjDNhkYmY-Gd9KuWj32fHIym1I7XPks5qRUB_gaJpZM4I8_Df>
> .
>
